src/Identity/Core/src/SecurityStampValidator.cs

```diff
@@ -23,16 +23,37 @@ public class SecurityStampValidator<TUser> : ISecurityStampValidator where TUser
     /// <param name="signInManager">The <see cref="SignInManager{TUser}"/>.</param>
     /// <param name="clock">The system clock.</param>
     /// <param name="logger">The logger.</param>
+    [Obsolete("ISystemClock is obsolete, use TimeProvider on SecurityStampValidatorOptions instead.")]
     public SecurityStampValidator(IOptions<SecurityStampValidatorOptions> options, SignInManager<TUser> signInManager, ISystemClock clock, ILoggerFactory logger)
     {
         ArgumentNullException.ThrowIfNull(options);
         ArgumentNullException.ThrowIfNull(signInManager);
         SignInManager = signInManager;
         Options = options.Value;
+        TimeProvider = Options.TimeProvider ?? TimeProvider.System;
         Clock = clock;
```

Comment:
Do we really want a `TimeProvider` and `Clock` that are out of sync?

```suggestion
        Clock = new SystemClock(TimeProvider);
```